### PR TITLE
Translations

### PR DESCRIPTION
**Pull request by almet**
Translation for the introduction and the permission article

### CODE REVIEW

File: content/2015.04.service-de-nuages-en.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+To build this service, we could have reused `Firefox sync`_ , after all it's
+a robust synchronisation service, built on top of `Cornice`_. But *Sync* hasn't
+been thought to guarantee strong data integrity and it was too complex to change that
+deeply.
+
+.. _Firefox Sync: https://github.com/mozilla-services/server-syncstorage
+.. _Cornice: http://cornice.readthedocs.org/
+
+We could also have created yet another application which exposes an API and
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Nit: to expose?

File: content/2015.04.service-de-nuages-en.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+persists data in a database.
+
+But, this little new team isn't here just by luck :)
+
+The «Daybed Team»
+-----------------
+
+We share a vision: a generic data storage service ! Maybe it brings to memories
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Nit: no space before punctuation in English (valid for the rest of other occurrences in this document)

File: content/2015.04.service-de-nuages-en.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+`a certain project named Daybed <https://daybed.io>`_ ?
+
+For client applications, Javascript or other, using this service should be as
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Nit: JavaScript (some people are really picky about this)

File: content/2015.04.service-de-nuages-en.rst [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+synchronize collections. As these problematics are recurring or even impossible
+to avoid, we decided to build on top of this synchronization protocol and to
+dissociate it from the readinglist project.
+
+The one-way mechanism
+---------------------
+
+As a first stone to build the cathedral, we created `Cliquet
+<https://cliquet.readthedocs.org>`_, which main idea is to have an
+implementation of this protocol in python, while factorizing all our good
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Nit: Python (cap P)

File: content/2015.04.service-de-nuages-en.rst [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+implementation!
+
+With *Cliquet*, the code of the readinglist service consists into defining
+a schema for the records and force some field values on certain calls. The size
+of the project is dramatically reduced to a dozen lines of code.
+
+As to the future generic data storage service, `the project
+<http://kinto.readthedocs.org>`_ just started but it really is!
+It can already be used as a storage backend in a *Cliquet* application, and
+this `implemented in 20 lines of code
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: "this _was_ implemented" I guess

File: content/2015.04.service-de-nuages-en.rst [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+<https://github.com/mozilla-services/kinto/blob/0.2.1/kinto/views/collection.py>`_!
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: General remark: we're heavily relying on exclamation marks across the document, maybe we should temper that a little?

File: content/2015.04.service-de-nuages-en.rst [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+Oh, and this time we will build features only from real use cases which come
+along the way. It might seem silly, but on *Daybed* we haven't seen it coming :)
```
Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Nit: which are coming along the way

Comment (n1k0) [inline content/2015.04.service-de-nuages-en.rst]: Also, instead of _we haven't seen it coming_, I'd favor _we never met that point_

File: content/2015.05.cliquet-permissions-en.rst [hunk 8]
```diff
@@ -0,0 +1,535 @@
+Handling permissions
+####################
+
+:date: 2015-05-12
+:summary: Where we clarify the vocabulary around permissions and explain what
+          we plan to implement for our generic data storage solution, Kinto.
+:slug: la-gestion-des-permissions
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:7]: I wish we could have the slug in English

File: content/2015.05.cliquet-permissions-en.rst [hunk 9]
```diff
@@ -0,0 +1,535 @@
...
+:lang: en
+
+When creating a generic data storage (Kinto), permissions handling is one of
+the big challenges: who should get access to what, and how should we define it?
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Maybe _how should we define these rules_?

File: content/2015.05.cliquet-permissions-en.rst [hunk 10]
```diff
@@ -0,0 +1,535 @@
...
+
+The problem at hand
+===================
+
+The problem we're facing is simple: data is stored online and we need a way to
+share this data with other people.
+
+Looking at the use cases, it's possible to split our users in two categories:
+
+- "Final" users (you);
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: End users

File: content/2015.05.cliquet-permissions-en.rst [hunk 11]
```diff
@@ -0,0 +1,535 @@
...
+- Applications which interact on their behalf.
+
+Different parties don't have the same rights: some should be able to read,
+others should write, yet another should be able to create new records, and all
+this permission handling should be fine grained: it should be possible to read
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Nit: fine-grained (dash)

File: content/2015.05.cliquet-permissions-en.rst [hunk 12]
```diff
@@ -0,0 +1,535 @@
...
+one record but not the other, for instance.
+
+We started from the fact that no available solution were offering an appealing
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: _no available solution was_

Edit: I've seen occurrences of _no available solutions were_, though I'm not sure

File: content/2015.05.cliquet-permissions-en.rst [hunk 13]
```diff
@@ -0,0 +1,535 @@
...
+
+A *principal*, is an entity that can be authenticated by a computer system.
+It is the actor who does the action. [#]_
+
+It could either be someone, a computer, a service or a group of any of such
+things. This is conceptually wider than the classical "user id*".
+
+Permissions are then associated to these *principals*.
+
+For instance, an user is identified in an unique way when he connects to the
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: _a user_, _a unique_ (http://english.stackexchange.com/a/105117)

Note: other occurrences were found in the document

File: content/2015.05.cliquet-permissions-en.rst [hunk 14]
```diff
@@ -0,0 +1,535 @@
...
+Tarek, Rémy and Alexis.
+
+
+Permissions delegation
+======================
+
+Imagine the following example, where an user stores two types of data online:
+
+- contacts ;
+- a list of tasks, which can be associated to her contacts.
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:105]: - a list of contacts;
- a list of tasks, which can be associated to her contacts.

Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:105]: Also, that's probably Shitstorm Compliant Ubernitpicking™, but I'm more comfortable with using _they/their_ instead of _she/her_; _they/their_ includes everybody by design.

Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:105]: Note that if you want to keep telling the story of an individual, we could name this user _Alice_, and that would make the whole thing absolutely fine.

File: content/2015.05.cliquet-permissions-en.rst [hunk 15]
```diff
@@ -0,0 +1,535 @@
...
+
+The user has all the rights on its data.
+
+However, she's using two applications which should have a restricted access:
+
+- One application to handle contacts, to which she wants to delegate the entire
+  management of the contacts: ``contacts:write`` ;
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Nit: repetition of contacts

File: content/2015.05.cliquet-permissions-en.rst [hunk 16]
```diff
@@ -0,0 +1,535 @@
...
+app should not be able to update existing contacts, just eventually creating
+new ones.
+
+She then needs a way to delegate some of her permissions to a third party (the
+app).
+
+That's precisely the role of `OAuth2 scopes
+<http://tools.ietf.org/html/rfc6749#page-23>`_.
+
+During the connection of an user, a window will ask her which access she want
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Same as for the previous document in this patch, _a user_

Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Also, _she wants_

File: content/2015.05.cliquet-permissions-en.rst [hunk 17]
```diff
@@ -0,0 +1,535 @@
...
++-----------------+---------------------------------------------------------+
+| Object          | Description                                             |
++=================+=========================================================+
+| **bucket**      | We can see them as *namespaces*. They allow to have     |
+|                 | different collections having the same name but stored   |
+|                 | in different *buckets* so that data are distinct.       |
++-----------------+---------------------------------------------------------+
+| **collection**  | A list of records.                                      |
++-----------------+---------------------------------------------------------+
+| **record**      | A record of a collection                                |
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Nit: _A record from a collection_, or _A record part of a collection_

File: content/2015.05.cliquet-permissions-en.rst [hunk 18]
```diff
@@ -0,0 +1,535 @@
...
+| Permission | Description                             |
++============+=========================================+
+| **read**   | The permission to read the content of   |
+|            | the object and all its children.        |
++------------+-----------------------------------------+
+| **write**  | The permission to modify and            |
+|            | administration an object and all its    |
+|            | children objects. The *write*           |
+|            | permission allows reading, modification |
+|            | and deletion of an objects, and the     |
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: _of objects_

File: content/2015.05.cliquet-permissions-en.rst [hunk 19]
```diff
@@ -0,0 +1,535 @@
...
+Actors connecting to the storage service can authenticate themselves.
+
+They then receive a list of *principals*:
+
+- ``Everyone``: the *principal* given to all actors (authenticated or not);
+- ``Authenticated``: the *principal* given to all authenticated actors;
+- A *principal* identifying the actor, for instance ``fxa:32aa95a474c984d41d395e2d0b614aa2``
+
+In order to avoid identifiers collisions, the actor *principal* is built from
+the authentication type she used (``system``, ``basic``, ``ipaddr``, ``hawk``,
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: As we previously mentioned that a principal can be a computer and was a generic abstraction on top of actors, _the authentication type it used_

Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: Hmm rereading it it's actually okay, scratch that.

File: content/2015.05.cliquet-permissions-en.rst [hunk 20]
```diff
@@ -0,0 +1,535 @@
...
+    {
+       "name": "Rémy",
+       "emails": ["remy@example.com"],
+       "phones": ["+330820800800"]
+    }
+
+    HTTP/1.1 307 Temporary Redirect
+    Location: /v1/buckets/fxa:49d02d55ad10973b7b9d0dc9eba7fdf0/collections/contacts/records
+
+Like so, it's possible for Alice to share her contacts with Bob. She just have
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:510]: So it seems you've actually named your user already :) we should do this at the beginning of the article.

File: content/2015.05.cliquet-permissions-en.rst [hunk 21]
```diff
@@ -0,0 +1,535 @@
...
+----------------------
+
+In *Kinto*, we defined a format to restrict permissions using OAuth2 scopes:
+``storage:<bucket_id>:<collection_id>:<permissions_list>``.
+
+Taking back the previous *tasks* example, it is possible for Bob to create
+a specific OAuth2 token with the following *scopes*:
+``profile storage:todolist:tasks:write storage:personal:contacts:read+records:create``
+
+Like so, even if Bob has the ``write`` permission on his contacts, the
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst:525]: on _her_ contacts, methinks (rather unclear tbh)

File: content/2015.05.cliquet-permissions-en.rst [hunk 22]
```diff
@@ -0,0 +1,535 @@
...
+application using this token will only be able to read the existing contacts
+and add new ones.
+
+One part of the complexity is to manage to present these *scopes* in a readable
```
Comment (n1k0) [inline content/2015.05.cliquet-permissions-en.rst]: _to manage presenting_

**Comment by n1k0** [comment]
Thanks for this, some nits here and there. r+ with comments addressed.

File: content/2015.04.service-de-nuages-en.rst (hunk 2 above)
Comment (almet) [inline content/2015.04.service-de-nuages-en.rst]: haha. I'm usually the one to nitpick about that ;)

File: content/2015.05.cliquet-permissions-en.rst (hunk 8 above)
Comment (almet) [inline content/2015.05.cliquet-permissions-en.rst:7]: This is due to how pelican handles translations (it adds a `-en` at the end of it) There might be a way to solve this, but not part of this PR.

File: content/2015.05.cliquet-permissions-en.rst (hunk 14 above)
Comment (almet) [inline content/2015.05.cliquet-permissions-en.rst:105]: This is indeed an attempt to use [gender neutral](https://english.stackexchange.com/questions/48/is-there-a-correct-gender-neutral-singular-pronoun-his-versus-her-versus) formulation.

From what I read there, it makes sense to use "their", but for me it doesn't convey the underlying I'm trying to. I'll update with "her or his".

Comment (almet) [inline content/2015.05.cliquet-permissions-en.rst:105]: Actually using "Alice" as an example as it seems even clearer to the reader.

File: content/2015.05.cliquet-permissions-en.rst (hunk 19 above)
Comment (almet) [inline content/2015.05.cliquet-permissions-en.rst]: Updated removing "she".

File: content/2015.05.cliquet-permissions-en.rst (hunk 21 above)
Comment (almet) [inline content/2015.05.cliquet-permissions-en.rst:525]: I believe it's "his".

**Comment by almet** [comment]
Thanks for the review. I've updated the PR so we're ready to merge.

**Comment by almet** [comment]
Hmm, before doing so, I think we should add a way to directly read in english from the index page.

Probably just adding a note saying that it's possible to have the index in english would be enough. Trying to do so now.
